xblockutils/studio_editable.py

```diff
@@ -323,3 +328,151 @@ def author_preview_view(self, context):
         not editing this block's children.
         """
         return self.student_view(context)
+
+
+class NestedXBlockSpec(object):
+    """
+    Class that allows detailed specification of allowed nested XBlocks. For use with
+    StudioContainerWithNestedXBlocksMixin.allowed_nested_blocks
+    """
+    def __init__(self, block, single_instance=False, disabled=False, disabled_reason=None):
+        self._block = block
+        self._single_instance = single_instance
+        self._disabled = disabled
+        self._disabled_reason = disabled_reason
+
+    @property
+    def category(self):
+        """ Block category - used as a computer-readable name of an XBlock """
+        return self._block.CATEGORY
+
+    @property
+    def label(self):
+        """ Block label - used as human-readable name of an XBlock """
+        return self._block.STUDIO_LABEL
+
+    @property
+    def single_instance(self):
+        """ If True, only allow single nested instance of Xblock """
+        return self._single_instance
+
+    @property
+    def disabled(self):
```

Comment:
@e-kolpakov Should we make a task for this as well? It seems like it may cause a problem if it's not eventually fixed.